components/app/index.tsx

```diff
@@ -6,28 +6,46 @@ import { ConfigContext } from '../config-provider';
 import useMessage from '../message/useMessage';
 import useModal from '../modal/useModal';
 import useNotification from '../notification/useNotification';
-import type { useAppProps } from './context';
-import AppContext from './context';
+import AppContext, { AppConfigContext } from './context';
+import type { AppConfig, useAppProps } from './context';
 import useStyle from './style';
 
 export type AppProps = {
   className?: string;
   rootClassName?: string;
   prefixCls?: string;
   children?: ReactNode;
-};
+} & AppConfig;
 
-const useApp = () => React.useContext<useAppProps>(AppContext);
+const useApp = () => React.useContext(AppContext);
 
-const App: React.FC<AppProps> & { useApp: () => useAppProps } = (props) => {
-  const { prefixCls: customizePrefixCls, children, className, rootClassName } = props;
+const App: React.FC<AppProps> & { useApp: typeof useApp } = (props) => {
+  const {
+    prefixCls: customizePrefixCls,
+    children,
+    className,
+    rootClassName,
+    message,
+    notification,
+  } = props;
   const { getPrefixCls } = useContext<ConfigConsumerProps>(ConfigContext);
   const prefixCls = getPrefixCls('app', customizePrefixCls);
   const [wrapSSR, hashId] = useStyle(prefixCls);
   const customClassName = classNames(hashId, prefixCls, className, rootClassName);
 
-  const [messageApi, messageContextHolder] = useMessage();
-  const [notificationApi, notificationContextHolder] = useNotification();
+  const appConfig = useContext(AppConfigContext);
+  const mergedAppConfig = React.useMemo<AppConfig>(
+    () => ({
+      message: { ...message, ...appConfig.message },
+      notification: { ...notification, ...appConfig.notification },
```

Comment:
```suggestion
      message: { ...appConfig.message, ...message  },
      notification: { ...appConfig.notification, ...notification, },
```
Current props should be respected. Add a test case for it?